protocol/peer/peer.go

```diff
@@ -31,6 +34,20 @@ type Peer struct {
 	lastPingNonce    uint64        // The nonce of the last ping we sent
 	lastPingTime     time.Time     // Time we sent last ping
 	lastPingDuration time.Duration // Time for last ping to return
+
+	isSelectedPeerRequested uint32
```

Comment:
It seems like here and in `selectedPeerRequestChan` you typo'd `Peer` instead of `tip`